ibis/pandas/aggcontext.py

```diff
@@ -501,7 +501,10 @@ def agg(
             # TODO: see if we can do this in the caller, when the context
             # is constructed rather than pulling out the data
             columns = group_by + order_by + [name]
-            indexed_by_ordering = frame[columns].set_index(order_by)
+            indexed_by_ordering = frame[columns]
```

Comment:
add a .copy() here to avoid mutations on the original

Reply:
Good catch. Let me add .

Reply:
Added